src/gc/unix/gcenv.unix.cpp

```diff
@@ -157,6 +160,60 @@ FOR_ALL_NUMA_FUNCTIONS
 #undef PER_FUNCTION_BLOCK
 #endif // HAVE_NUMA_H
 
+#if defined(__linux__) && HAVE_NUMA_H
+static bool ShouldOpenLibNuma()
+{
+    // This is a simple heuristic to determine if libnuma.so should be opened.  There's
+    // no point in linking and resolving everything in this library if we're running on
+    // a system that's not NUMA-capable.
+    int fd;
+
+    while (true)
+    {
+        fd = open("/sys/devices/system/node/possible", O_RDONLY | O_CLOEXEC);
+
+        if (fd == -1)
+        {
+            if (errno == EINTR)
+            {
+                continue;
+            }
+
+            // sysfs might not be mounted, not available, or the interface might have
+            // changed.  Return `true' here so NUMASupportInitialize() can try initializing
+            // NUMA support with libnuma.
+            return true;
+        }
+
+        break;
+    }
+
+    while (true)
+    {
+        char buffer[32];
+        ssize_t bytesRead = read(fd, buffer, 32);
+
+        if (bytesRead == -1 && errno == EINTR)
+            continue;
+
+        if (!strncmp(buffer, "0\n", bytesRead))
+        {
+            // Only one node, so avoid linking libnuma.so.
+            close(fd);
```

Comment:
A nit - could you please have just a single close(fd)? E.g. by moving it before the `if`.

Reply:
Done. Also fixed the case where bytesRead could be -1 with any other errno (which would be passed to strncmp()).